src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -696,6 +696,14 @@ private async Task<bool> EnumerateArtifact(IEnumeratedArtifact artifact, TContex
 
             if (IsOpcArtifact(artifact, filePath, globalContext))
             {
+                // Check recursion depth to prevent stack overflow
+                if (recursionDepth >= globalContext.MaxArchiveRecursionDepth)
+                {
+                    string reason = $"archive nesting exceeded maximum depth of {globalContext.MaxArchiveRecursionDepth}";
```

Comment:
is this casing consistent with other messages? do you have an example of a fully formed note message created by this code path?